### PR TITLE
Add branch representation and API calls

### PR DESCRIPTION
**Pull request by sapk**
Needed for https://github.com/gogits/gogs/issues/2410

The representation should be near Github API :
https://developer.github.com/v3/repos/#list-branches

### CODE REVIEW

**Comment by Unknwon** [comment]
Thanks your PR... but Gogs does not provide this info at all currently... which means no use of this APIs.

**Comment by sapk** [comment]
The goal is to add it like https://github.com/gogits/gogs/blob/9cd16c5b12d093dd694e146d6c845ee4c405c774/routers/api/v1/user/keys.go.
Maybe I should do it in 2 step ?
- Add the representation (structure)
- Add api call after the api is implemented in gogs

**Comment by sapk** [comment]
I coudl also only return the name attribute for the API response and don't have to rely on API struc representation.

**Comment by Unknwon** [comment]
This PR will remain open until server side is implemented :D

File: repo_branches.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package gogs
+
+type Branch struct {
+	name      string         `json:"name"`
```
Comment (Unknwon) [inline repo_branches.go]: The first letter of fields must be upper case.

**Comment by sapk** [comment]
Related wiki page (draft): https://gist.github.com/sapk/df64347ff218baf4a277
the url link is not implemented since there is no API for commit like https://developer.github.com/v3/repos/commits/

**Comment by bkcsoft** [comment]
@sapk is that *really* a **POST** to **GET** data? :smiley:

**Comment by sapk** [comment]
Sorry bad copy-paste in the draft ^^
Fixed.
@Unknown already fix it in the official wiki : https://github.com/gogits/go-gogs-client/wiki/Repositories

**Pull request merged**